packages/realm_dart/lib/src/init.dart

```diff
@@ -65,26 +58,39 @@ String get _exeDirName => p.dirname(Platform.resolvedExecutable);
 
 DynamicLibrary _openRealmLib() {
   final libName = _getLibName(realmBinaryName);
-  final root = _getNearestProjectRoot(Platform.script.path) ?? _getNearestProjectRoot(p.current);
-
-  // Try to open lib from various candidate paths
-  List<String> errMessages = [];
-  for (final open in [
-    () => _open(libName), // just ask OS..
-    () => _open(p.join(_exeDirName, libName)), // try finding it next to the executable
-    if (root != null) () => _open(p.join(root, 'binary', Platform.operatingSystem, libName)), // try finding it relative to project
-    () => _open(_getBinaryPath(libName)), // find it where it is installed by plugin
-  ]) {
+
+  DynamicLibrary? tryOpen(String path) {
     try {
-      return open();
-    } on Error catch (e) {
-      errMessages.add(e.toString());
+      return DynamicLibrary.open(path);
+    } on Error catch (_) {
+      return null;
     }
   }
-  throw RealmError(errMessages.join('\n'));
-}
 
-DynamicLibrary _open(String lib) => DynamicLibrary.open(lib);
+  Never throwError(Iterable<String> candidatePaths) {
+    final message = 'Could not open $libName at any of the following paths:\n${candidatePaths.join('\n')}'
+        '${isFlutterPlatform ? '\nDid you forget to add a dependency on realm package?' : 'Did you forget to run `dart run realm_dart install`?'}';
```

Comment:
Perhaps this could make it somewhat easier to parse.

(P.S. I think PR description examples showed an earlier iteration)
```suggestion
    final message = 'Could not open $libName at any of the following paths:\n- ${candidatePaths.join('\n- ')}'
        '${isFlutterPlatform ? '\nHint: Did you forget to add a dependency on realm package?' : '\nHint: Did you forget to run `dart run realm_dart install`?'}';
```

Reply:
Hmm.. I think I'm missing a push 😬

Reply:
Added the '- ' prefix, the 'Hint: ' prefix, and wrapped paths in ""